### PR TITLE
Feature: Add a `manager` role who can extend a pool

### PR DESCRIPTION
**Pull request by sujine2**
The admin registers a different address as a manager, and the admin can set the duration and rewardPerSecond of the next staking pool.

### CODE REVIEW

File: test/claim.test.ts [hunk 1]
```diff
@@ -263,6 +263,70 @@ describe('StakingPool.claim', () => {
       expect(poolDataAfterClaim).to.be.equalPoolData(expectedPoolDataClaim);
       expect(userDataAfterClaim).to.be.equalUserData(expectedUserDataClaim);
     });
+
+    context('amdin set bob as manager', async () => {
```
Comment (jaeboklee-modo) [inline test/claim.test.ts]: typo: admin!

File: test/claim.test.ts [hunk 2]
```diff
@@ -263,6 +263,70 @@ describe('StakingPool.claim', () => {
...
+      it('bob becomes manager and call extend pool and alice claim', async () => {
+        await actions.faucetAndApproveTarget(bob, RAY);
+        const poolDataBefore = await getPoolData(testEnv);
+        const userDataBefore = await getUserData(testEnv, alice);
+        await testEnv.stakingPool.setManager(bob.address);
+  
+        advanceTimeTo(secondRoundStartTimestamp)
```
Comment (jaeboklee-modo) [inline test/claim.test.ts]: await should be added?

File: test/settings.test.ts [hunk 3]
```diff
@@ -93,18 +93,33 @@ describe('StakingPool.settings', () => {
...
+      ).to.be.revertedWith('OnlyAdmin');
+      */
+    });
+
+    it('success when alice is set up as a manger by amdin and', async () => {
+      await expect(testEnv.stakingPool.connect(depositor).setManager(depositor.address))
+      .to.be.revertedWith(
+        'OnlyAdmin'
+      );
+      await testEnv.stakingPool.connect(deployer).setManager(depositor.address);
```
Comment (jaeboklee-modo) [inline test/settings.test.ts]: What are these lines for?

**Review by jaeboklee-modo — CHANGES_REQUESTED** [review]
Some of the `advanceTimeTo` function are not called asynchronously. Does it work well?
And nit: amdin -> admin! :)

File: test/settings.test.ts (hunk 3 above)
Comment (sujine2) [reply test/settings.test.ts]: This is to check that the setManager function can be called if admin

Comment (jaeboklee-modo) [reply test/settings.test.ts]: All right, then I think the check process should be added; Is this function works as intended? Like sets storage or emits an event.

File: contracts/StakingPoolV2.sol [hunk 4]
```diff
@@ -13,12 +14,14 @@ import '@openzeppelin/contracts/access/Ownable.sol';
...
 /// @author Elysia
-contract StakingPoolV2 is IStakingPoolV2, StakedElyfiToken, Ownable {
+contract StakingPoolV2 is IStakingPoolV2, StakedElyfiToken, Ownable, AccessControl {
   using StakingPoolLogicV2 for PoolData;
 
   constructor(IERC20 stakingAsset_, IERC20 rewardAsset_) StakedElyfiToken(stakingAsset_) {
     stakingAsset = stakingAsset_;
     rewardAsset = rewardAsset_;
+    _admin = msg.sender;
+    _setupRole(MANAGER_ROLE, msg.sender);
```
Comment (qpzm) [inline contracts/StakingPoolV2.sol]: Are there any way to revoke the manager role? It seems that there is no role admin for `MANAGER_ROLE` or `DEFAULT_ROLE_ADMIN`.

File: contracts/StakingPoolV2.sol [hunk 5]
```diff
@@ -231,6 +235,10 @@ contract StakingPoolV2 is IStakingPoolV2, StakedElyfiToken, Ownable {
     SafeERC20.safeTransfer(rewardAsset, _msgSender(), residueAmount);
   }
 
+  function setManager(address managerAddr) external onlyOwner {
+    _setupRole(MANAGER_ROLE, managerAddr);
+  }
```
Comment (qpzm) [inline contracts/StakingPoolV2.sol]: If you set the `DEFAULT_ROLE_ADMIN` in the constructor by `_setupRole(DEFAULT_ADMIN_ROLE, msg.sender);`, you can just use `grantRole` of the `AccessControl` contract without this function.

Reference
https://docs.openzeppelin.com/contracts/4.x/access-control#granting-and-revoking

Comment (jaeboklee-modo) [reply contracts/StakingPoolV2.sol]: 👍

File: test/withdraw.test.ts [hunk 6]
```diff
@@ -362,6 +362,74 @@ describe('StakingPool.withdraw', () => {
         expect(poolDataAfterWithdraw).eql(expectedPoolDataWithdraw);
         expect(userDataAfterWithdraw).eql(expectedUserDataWithdraw);
       });
+
+
+      context('amdin set bob as manager', async () => {
+        it('bob becomes manager and call extend pool and alice withdraw', async () => {
```
Comment (qpzm) [inline test/withdraw.test.ts:368]: Why is 'alice withdraws' part necessary? I think the test in `settings.test.ts` is enough: when alice is set up as a manger by admin, she can extend the pool.

**Pull request merged**